### DISCUSSION TITLE
How not to blur the volume while rotating in volume rendering?

### DISCUSSION THREAD

**Discussion by jihankim** [pyvista/pyvista#6749] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Description

How can I turn off the blur effect while rotating the volume in volume rendering in pyvista?

I've looked up and I found this: https://discourse.vtk.org/t/in-volume-rendering-how-to-set-the-rendering-result-to-not-blur-when-rotating-the-camera/11294

Seems there's some way to turn it off on native VTK, but I couldn't find anyway in pyvista.

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Tue Oct 29 20:24:10 2024 Pacific Daylight Time
                OS : Windows (10 10.0.19045 SP0 Multiprocessor Free)
            CPU(s) : 12
           Machine : AMD64
      Architecture : 64bit
               RAM : 15.3 GiB
       Environment : IPython
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon(TM) Graphics
       GPU Version : 4.6.0 Compatibility Profile Context 24.7.1.240618
  MathText Support : False
  Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64
  bit (AMD64)]
           pyvista : 0.44.1
               vtk : 9.3.1
             numpy : 1.26.4
        matplotlib : 3.8.2
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 10.4.0
           imageio : 2.33.1
             PyQt5 : 5.15.9
           IPython : 8.9.0
        ipywidgets : 8.1.1
             scipy : 1.11.4
              tqdm : 4.66.1
        jupyterlab : 4.0.8
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

### Screenshots

Image gets blurry when rotating
![blur](https://github.com/user-attachments/assets/e6553892-420d-4b1d-922b-68765fb46d7b)

Still image looks okay and sharp
![okay](https://github.com/user-attachments/assets/40a3a264-492b-4c4d-b866-5a17213ed50d)

**Comment by banesullivan** [comment, marked as answer]
We don't have this handled in PyVista directly yet but this is something that has been bugging me as well. I'm not sure if this following snippet does it or not, but you can adjust the parameters of the underly VTK mapper through the returned actor:

```py
import pyvista as pv
vol = pv.Wavelet()

pl = pv.Plotter()
actor = pl.add_volume(vol)
actor.mapper.AutoAdjustSampleDistancesOff()
actor.mapper.InteractiveAdjustSampleDistancesOff()
pl.show()
```

I'll add this to [the backlog](https://github.com/pyvista/pyvista/issues/4626) of volume rendering issues we plan to address in an upcoming overhaul to volume rendering

**Reply by jihankim** [reply]
Thank you so much!
Although I got error "AttributeError: 'FixedPointVolumeRayCastMapper' object has no attribute 'InteractiveAdjustSampleDistancesOff'" on "actor.mapper.InteractiveAdjustSampleDistancesOff()", it still works after I removed that part. "actor.mapper.AutoAdjustSampleDistancesOff()" still does exactly what I need! :)
